backend/src/posts/dtos/searchPost.dto.ts

```diff
@@ -38,4 +39,11 @@ export default class SearchPostDto {
   @IsOptional()
   @IsDateString()
   endDateTime: Date;
+
+  @IsOptional()
+  @IsBoolean()
+  @Transform((val) =>
+    val.value === 'true' ? true : val.value === 'false' ? false : '',
```

Comment:
Nit: Nested ternary operators is abit hard to read. Might want to consider changing to writing if else for one of the condition